### PR TITLE
Closes # 206 - Add functionality to contact us form

### PR DESCRIPTION
**Pull request by OleksandrBartosh**
Added validation to form, refactored and optimized code, changed footer height, removed large whitespaces on /home page, changed CSS styles

### CODE REVIEW

**Review by antonbakalets — APPROVED** [review]

File: src/app/page/home/home.component.ts [hunk 1]
```diff
@@ -191,8 +195,14 @@ export class HomeComponent implements OnInit {
...
   sendEmailMessage(form: any) {
-    this.mainService.sendEmailMessage(form).subscribe();
-    this.clearInputFields();
-    this.emailSended = true;
+    this.contactForm = new ContuctUsEmail(form.name, form.email, form.subject, form.text);
+    this.mainService.sendEmailMessage(form).subscribe(response => {
+      if (response.ok) {
+        this.emailSent = true;
+        this.contactUsForm.reset();
+      }else {
```
Comment (Maks-Chornyi) [inline src/app/page/home/home.component.ts]: space

File: src/app/page/home/home.css [hunk 2]
```diff
@@ -253,6 +243,33 @@
...
+  font-weight: 600;
+  font-size: large;
+  padding: 1rem 1.25rem;
+  text-align: center;
+}
+
+.alert-success{
+  border: none;
+  background-color: #ffc91b;
+
```
Comment (Maks-Chornyi) [inline src/app/page/home/home.css]: delete empty line

File: src/app/page/home/home.css [hunk 3]
```diff
@@ -287,6 +296,10 @@
     flex-direction: column;
   }
 
+  .container-form > .col-5, .col-7{
```
Comment (Maks-Chornyi) [inline src/app/page/home/home.css]: add space before curly brace

File: src/app/page/home/home.css [hunk 4]
```diff
@@ -253,6 +243,33 @@
...
+  text-align: center;
+}
+
+.alert-success{
+  border: none;
+  background-color: #ffc91b;
+
+}
+
+.alert-danger{
```
Comment (Maks-Chornyi) [inline src/app/page/home/home.css]: need one space

**Review by Maks-Chornyi — APPROVED** [review]

File: src/app/page/home/home.component.ts [hunk 5]
```diff
@@ -32,11 +33,10 @@ export class HomeComponent implements OnInit {
   public subscriptionButtonText: string[] = [];
   coursesWithStatus: CourseLinkWithStatus[] = [];
   private coursesIdOfTheUser: number[] = [];
-  name: string;
-  email: string;
-  subject: string;
-  text: string;
-  emailSended: boolean;
+  contactUsForm: FormGroup;
+  contactForm: ContuctUsEmail;
```
Comment (maxturko) [inline src/app/page/home/home.component.ts]: typo in variable name ContuctUsEmail

**Comment by maxturko** [comment]
When the server is not working I do not get any error message, do you think it should work this way?

**Comment by maxturko** [comment]
Your messages do not look very friendly in my opinion. I don't know if it is very important or not but something like this looks better to me
https://cdn-images-1.medium.com/max/800/1*zsxo1iik1sFKCGDChptztw.png

**Review by maxturko — APPROVED** [review]

**Review by Maks-Chornyi — APPROVED** [review]

**Review by hlibOstapenko — APPROVED** [review]

**Review by OleksandrMazur — APPROVED** [review]

**Pull request merged**